### DISCUSSION TITLE
still Json Parse error

### DISCUSSION THREAD

**Discussion by Rasb74** [rern/rAudio#1328] [Bug Reports]
```
Errors: JSON Parse error: Unable to parse JSON string CopyX{ 
  "player"       : "mpd"
, "btreceiver"   : false
, "card"         : 1
, "consume"      : false
, "control"      : "SMSL USB AUDIO"
, "counts"       : {
  "album": 5819,
  "albumartist": 2824,
  "artist": 4957,
  "composer": 5740,
  "conductor": 353,
  "dabradio": 0,
  "date": 1104,
  "genre": 349,
  "latest": 0,
  "nas": 3,
  "playlists": 0,
  "sd": 0,
  "song": 60698,
  "usb": 0,
  "webradio": 32
}
, "icon"         : ""
, "librandom"    : false
, "lyrics"       : true
, "relays"       : false
, "relayson"     : false
, "scrobble"     : true
, "shareddata"   : false
, "snapclient"   : false
, "stoptimer"    : false
, "updateaddons" : false
, "updating_db"  : false
, "updatingdab"  : false
, "volume"       : 100
, "volumemute"   : 0
, "webradio"     : false
, "display"          : [
    true,
    false,
    false
]
, "audiocd"          : false
, "camilladsp"       : false
, "color"            : ""
, "dabradio"         : false
, "equalizer"        : false
, "multiraudio"      : false
, "relays"           : false
, "screenoff"        : true
, "snapclient"       : false
, "volumenone"       : true
}
, "repeat" : false
, "random" : false
, "single" : false
, "file"      : "NAS/Pop Rock/Aretha Franklin - Laughing On the Outside (Expanded Edition) (1963, 2011, 2017)/10. Where Are You.flac"
, "pllength"  : 17
, "song"      : 10
, "state"     : "play"
, "timestamp" : 1685296857896
, "stream" : false
, "Album"  : "Laughing on the Outside"
, "Artist" : "Aretha Franklin"
, "Time"   : 234
, "Title"  : "Where Are You?"
, "ext"      : "FLAC"
, "coverart" : ""
, "icon"     : ""
, "sampling" : "11/17 • 24 bit 96.0 kHz 2.91 Mbit/s • FLAC"
, "elapsed"  : 199
, "login" : false }```

**Comment by ceejayemm** [comment]
Is this similar to #1308 ?

Chris

**Comment by voterpublic** [comment]
After update 20230528 I get this error & no longer able to accces home pages. thanks

```
Errors: JSON.parse: bad escaped character at line 99 column 19 of the JSON data 
{ 
  "player"      X : "mpd"
, "btreceiver"   : false
, "card"         : 0
, "consume"      : false
, "control"      : "Digital"
, "counts"       : {
  "album": 320,
  "albumartist": 129,
  "artist": 315,
  "composer": 69,
  "conductor": 0,
  "dabradio": 0,
  "date": 177,
  "genre": 50,
  "latest": 0,
  "nas": 1,
  "playlists": 0,
  "sd": 0,
  "song": 3785,
  "usb": 0,
  "webradio": 4
}
, "icon"         : ""
, "librandom"    : true
, "lyrics"       : false
, "relays"       : false
, "relayson"     : false
, "scrobble"     : false
, "shareddata"   : false
, "snapclient"   : false
, "stoptimer"    : false
, "updateaddons" : false
, "updating_db"  : false
, "updatingdab"  : false
, "volume"       : 100
, "volumemute"   : 0
, "webradio"     : false
, "display"          : {
  "album": true,
  "albumartist": true,
  "albumbyartist": false,
  "artist": true,
  "audiocdplclear": false,
  "backonleft": true,
  "bars": true,
  "barsalways": false,
  "buttons": true,
  "camilladsp": false,
  "composer": true,
  "conductor": false,
  "count": true,
  "cover": true,
  "covervu": false,
  "dabradio": false,
  "date": false,
  "dsp": false,
  "fixedcover": true,
  "genre": true,
  "hidecover": false,
  "label": true,
  "latest": true,
  "multiraudio": false,
  "nas": true,
  "noswipe": false,
  "playbackswitch": true,
  "playlists": true,
  "plclear": true,
  "plsimilar": true,
  "radioelapsed": false,
  "sd": true,
  "tapaddplay": false,
  "tapreplaceplay": false,
  "time": true,
  "usb": true,
  "volume": true,
  "vumeter": false,
  "webradio": true
, "audiocd"          : false
, "camilladsp"       : false
, "color"            : ""
, "dabradio"         : false
, "equalizer"        : false
, "multiraudio"      : false
, "relays"           : false
, "screenoff"        : false
, "snapclient"       : false
, "volumenone"       : true
}
, "repeat" : false
, "random" : false
, "single" : false
, "file"      : "NAS/Do.The.Music/The Radio Birdman/'74_'78 collection/01-Aloha steve & danno.mp3"
, "pllength"  : 64
, "song"      : 58
, "state"     : "stop"
, "timestamp" : 1685318191126
, "stream" : false
, "Album"  : "'74\'78 collection"
, "Artist" : "The Radio Birdman"
, "Time"   : 236
, "Title"  : "Aloha steve & danno"
, "ext"      : "MP3"
, "coverart" : ""
, "icon"     : ""
, "sampling" : "59/64 • 44.1 kHz 320 kbit/s • MP3"
, "elapsed"  : 0
, "coverart" : ""
, "login" : false }```

**Reply by voterpublic** [reply]
Hi,
OK i have tried to install to existing & new sd card version rAudio-64bit-20230521 5 times now.
Tried to restore from backup & clean setup. Everytime library or dac change triggered the above error happens.
Back to version rAudio-64bit-20230212 stable, no issues.
Wondering if this is caused by the DAC hat eeprom issue? Tried to flash eeprom but no change error reoccurred.
Ta
Pi3B HiFiBerry DAC + PRO

**Reply by rern** [reply]
Your `Errors: JSON.parse` was caused by `, "Album"  : "'74\'78 collection"`.
Remove the `\` from album tag and try again.

**Reply by voterpublic** [reply]
'74_'78 collection is the actual folder name. No \ but _  . Change it to blank?
Ta very much Rern

**Reply by voterpublic** [reply]
I have installed the latest however it is still claiming there is an update. Also from the system screenshot attached the version is 20230310. Is that right? Should i update before adding my library?
PS. I have updated tag to blank.
![Screenshot 2023-05-30 130621](https://github.com/rern/rAudio/assets/112673643/cddaba5e-ddd6-4092-947c-dfd53fe476e0)

**Reply by voterpublic** [reply]
Error on library update

```
Errors: Unexpected token 'U', "Updating D"... is not valid JSON CopyXUpdating DB (#1) ...
{ 
  "player"       : "mpd"
, "btreceiver"   : false
, "card"         : 0
, "consume"      : false
, "control"      : "Digital"
, "counts"       : {
  "playlists" : 0
, "webradio"  : 4
}
, "icon"         : ""
, "librandom"    : false
, "lyrics"       : false
, "relays"       : false
, "relayson"     : false
, "scrobble"     : false
, "shareddata"   : false
, "snapclient"   : false
, "stoptimer"    : false
, "updateaddons" : true
, "updating_db"  : true
, "updatingdab"  : false
, "volume"       : 100
, "volumemute"   : 0
, "repeat" : false
, "random" : false
, "single" : false
, "file"      : "https://stream.radioparadise.com/mellow-flac"
, "pllength"  : 1
, "song"      : 0
, "timestamp" : 1685443355602
, "stream" : true
, "Album"        : ""
, "Artist"       : ""
, "stationcover" : "/data/webradio/img/https:||stream.radioparadise.com|mellow-flac.jpg"
, "Name"         : "Radio Paradise Mellow Mix (flac) - playlist: radioparadise.com"
, "state"        : "stop"
, "station"      : "Radio Paradise - Mellow Mix"
, "Time"         : false
, "Title"        : ""
, "webradio"     : true
, "ext"      : "Radio"
, "coverart" : ""
, "icon"     : "radioparadise"
, "sampling" : "1/1 • 44.1 kHz 192 kbit/s"
, "elapsed"  : 0
, "coverart" : ""
, "login" : false }```

**Reply by voterpublic** [reply]
New install with latest image. Update via addon immediately.
Great success!
Many thanks for your time Rern and all.

**Comment by findnix** [comment]
Same here ...

The directories  `/srv/http/assets/js/plugin` and `/srv/http/assets/css/plugin` where missing after upgrade.
Reconstructed them fom another, not updated player

After that see this in browser ... 
`Errors: Unexpected non-whitespace character after JSON at position 1032`

corrospending error message from journalctl:
`May 29 12:23:54 rAudio-WE nginx[436]: 2023/05/29 12:23:54 [error] 436#436: *27 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $addons in /srv/http/common.php on line 106PHP message: PHP Warning:  Undefined variable $addonsprogress in /srv/http/common.php on line 106PHP message: PHP Warning:  Undefined variable $guide in /srv/http/common.php on line 106PHP message: PHP Warning:  Undefined variable $relays in /srv/http/common.php on line 106PHP message: PHP Warning:  Undefined variable $addons in /srv/http/common.php on line 109PHP message: PHP Warning:  Undefined variable $addonsprogress in /srv/http/common.php on line 109PHP message: PHP Warning:  Undefined variable $guide in /srv/http/common.php on line 109PHP message: PHP Warning:  Undefined variable $addonsprogress in /srv/http/common.php on line 112PHP message: PHP Warning:  Undefined variable $guide in /srv/http/common.php on line 112" while reading response header from upstream, client: 10.100.1.160, server: localhost, request: "GET / HTTP/1.1", upstream:"fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "10.100.1.164"`

mpd is working fine, so i am still able to control the player from outside.

**Reply by rern** [reply]
Please try the [latest image file](https://github.com/rern/rAudio/releases/tag/i20230521).

**Comment by surfninja2** [comment]
The latest update seemed to brick both of my rAudio instances.

I loaded the latest image file pre **rern** above (20230521) and I'm getting the same JSON error.  


I don't mind starting from scratch (again) and can help with errors.

I'd like to try 20230528 but I don't know how to create an image file.  is this possible to create an img file from the latest source?  I using linux mint and do have RP Imager installed.

**Reply by ceejayemm** [reply]
Flash i20230521 (thelatest image) to an SD card. Once this boots you should be able to update to 20230528 via the Addons screen. Only Rern can create a new image. 

Chris

**Reply by surfninja2** [reply]
I was able to update to 20230528 and at least the Radio Paradise web radio is now playing.  rebuilding library...

thanks **Rern** and **Chris**.

BTW I was able to get my other RP working.

I ssh'd into the raspberry pi and 'curl' the install script.  like:
Audio-1:~  #  curl -sfLO https://github.com/rern/rAudio-1/raw/main/install.sh
rAudio-1:~  #  chmod 755 install.sh
rAudio-1:~  #  ./install.sh "r1
update
20230528"


this got it alive again.  i had to re-update (Addons->Update).  you might be able to replace the **20230528** with **main**.  I just ran the update again.

**Reply by kw-sudo** [reply]
Due to the current bugs/issues, I went back to the i20230521 image. As for the "json parse error", here's what I can reproduce:

After a clean install from the scratch, I was able to do all the settings except for the 'playlist'. After deselecting all items there, the error message came after reboot.


![json parse error](https://github.com/rern/rAudio/assets/104136223/dfbb65ef-4dcf-4a91-80d5-486772042d52)


The next time after first run (I did nothing else), first I checked the playlist settings and did a reboot: bamm! json parse error. :-/

**Reply by voterpublic** [reply]
Yeah I did that and got the same. Just use latest image on reformatted sd card. Boot and do nothing to any settings. Check addon for update. Do the update. After reboot add your settings from backup. Should be ok
